#!/usr/bin/env python
PACKAGE = "cloud_factory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("width", double_t, 0, "Adjust the width of the plane", 5.0, 1.0, 10.0)
gen.add("height", double_t, 0, "Adjust the height of the plane", 5.0, 1.0, 10.0)
gen.add("resolution", double_t, 0, "Adjust the resolution of the plane", 0.01, 0.01, 1.0)
gen.add("step_height_0", double_t, 0, "Adjust the height of the step0", 0.1, 0.01, 10.0)
gen.add("step_height_1", double_t, 0, "Adjust the height of the step1", 0.2, 0.01, 10.0)
gen.add("step_height_2", double_t, 0, "Adjust the height of the step2", 0.3, 0.01, 10.0)
gen.add("step_height_3", double_t, 0, "Adjust the height of the step3", 0.4, 0.01, 10.0)
gen.add("n_points_step", int_t, 0, "Adjust the number of points up the step", 10, 1, 100)
gen.add("noise_points", double_t, 0, "Add a bit of random noise to z-koordinate of the points ", 0.001, 0.00001, 0.05)
exit(gen.generate(PACKAGE, "cloud_factory", "Steps4"))